src/lib.rs

```diff
@@ -66,6 +87,53 @@ impl VehiclePropertyValue {
         }
     }
 
+    fn from_prop(
+        vhal_prop_type: &VehiclePropertyType,
+        prop_value: &VehicleHalProto::VehiclePropValue,
+    ) -> Result<Self> {
+        Ok(match *vhal_prop_type {
+            VehiclePropertyType::STRING => Self::String(
+                prop_value
+                    .string_value
+                    .as_ref()
+                    .ok_or(VhalError::PropertyTypeError)?
+                    .to_owned(),
+            ),
+            VehiclePropertyType::BYTES => Self::Bytes(
+                prop_value
+                    .bytes_value
+                    .as_ref()
+                    .ok_or(VhalError::PropertyTypeError)?
+                    .to_vec(),
+            ),
+            VehiclePropertyType::BOOLEAN => Self::Int32(
```

Comment:
Why not to `bool`?

Reply:
This will ultimately be sent to the network as an int32: https://android.googlesource.com/platform/packages/services/Car/+/refs/heads/main/tools/emulator/vhal_emulator_test.py#117

Reply:
OK.
Not arguing with you :)
Though, using boolean
(1) might be more easier to understand for someone reading the code (while the inner implementation can do whatever it likes)
and
(2) will not assume/require knowledge of internal implementation that might theoretically change in the future to be able to send a boolean.
Again, maybe I'm wrong :)

Reply:
Well, maybe. We would have to add an enum variant and handle the case, ultimately transforming it to int32 when setting the `VehiclePropValue`. I'll open an issue with this dicussion.